components/common/src/main/java/com/hotels/styx/common/content/ContentPublisher.java

```diff
@@ -0,0 +1,53 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.common.content;
+
+import com.hotels.styx.api.Buffer;
+import io.netty.channel.EventLoop;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * A publisher to wrap the FlowControllingHttpContentProducer FSM and perform subscription operations via a Netty EventLoop.
+ */
+public final class ContentPublisher implements Publisher<Buffer> {
```

Comment:
Perhaps we should rename this to `ByteStreamPublisher`?  Because it is the ByteStream publisher for which this class produces content. That might make it easier for other devs make the connection.

Reply:
That couples it deeply to the ByteStream class, which it knows nothing about.  It only knows about the FlowControllingHttpContentProducer and even that doesn't know about ByteStream.  However, I get your point, the name is not very meaningful, so I have renamed it to FlowControllingPublisher.  Is that more helpful?

Reply:
Hi. It was just a suggestion. I'm not too opinionated about this. But between `FlowControllingPublisher` and `ContentPublisher`, I would prefer the original because it is shorter and simpler :-)